src/Components/Endpoints/src/RazorComponentEndpointInvoker.cs

```diff
@@ -46,8 +46,14 @@ private async Task RenderComponentCore(HttpContext context)
 
         var endpoint = context.GetEndpoint() ?? throw new InvalidOperationException($"An endpoint must be set on the '{nameof(HttpContext)}'.");
 
+        var componentTypeMetadata = endpoint.Metadata.GetRequiredMetadata<ComponentTypeMetadata>();
         var rootComponent = endpoint.Metadata.GetRequiredMetadata<RootComponentMetadata>().Type;
-        var pageComponent = endpoint.Metadata.GetRequiredMetadata<ComponentTypeMetadata>().Type;
+        var pageComponent = componentTypeMetadata.Type;
+
+        if (componentTypeMetadata.IsStaticRoute)
+        {
+            _renderer.SuppressRootComponentRenderModes();
+        }
```

Comment:
This is the bit where we take the endpoint metadata and turn it into a behavior on the renderer.